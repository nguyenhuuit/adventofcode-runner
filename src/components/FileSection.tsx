import chalk from 'chalk';

import React from 'react';

import { Box, Text } from 'ink';

interface FileSectionProps {
  solutionFileName: string;
  solutionFileSize: number;
  inputFileName: string;
  inputFileSize: number;
}

const FileSection = ({
  solutionFileName,
  solutionFileSize,
  inputFileName,
  inputFileSize,
}: FileSectionProps) => {
  return (
    <Box width={100}>
      <Box flexDirection="column" width={70}>
        <Text>{`Solution file: ${chalk.bold(chalk.magentaBright(solutionFileName.slice(2)))} ${solutionFileSize} bytes`}</Text>
        <Text>{`   Input file: ${chalk.bold(chalk.magentaBright(inputFileName.slice(2)))} ${inputFileSize} bytes`}</Text>
      </Box>
    </Box>
  );
};

export default FileSection;
