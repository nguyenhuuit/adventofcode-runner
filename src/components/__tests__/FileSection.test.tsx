import { render } from '@testing-library/react';

import React from 'react';

import FileSection from '../FileSection';

interface MockBoxProps {
  children: React.ReactNode;
  [key: string]: unknown;
}

interface MockTextProps {
  children: React.ReactNode;
}

jest.mock('ink', () => ({
  Box: ({ children, ...props }: MockBoxProps) => (
    <div data-testid="box" {...props}>
      {children}
    </div>
  ),
  Text: ({ children }: MockTextProps) => <span data-testid="text">{children}</span>,
}));

jest.mock('chalk', () => ({
  bold: (text: string) => text,
  magentaBright: (text: string) => text,
}));

describe('FileSection', () => {
  it('renders correctly with file info', () => {
    const { container } = render(
      <FileSection
        solutionFileName="./solution.ts"
        solutionFileSize={123}
        inputFileName="./input.txt"
        inputFileSize={456}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with empty file sizes', () => {
    const { container } = render(
      <FileSection
        solutionFileName="./solution.ts"
        solutionFileSize={0}
        inputFileName="./input.txt"
        inputFileSize={0}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
