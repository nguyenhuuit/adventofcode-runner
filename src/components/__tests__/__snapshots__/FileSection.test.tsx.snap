// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileSection renders correctly with empty file sizes 1`] = `
<div>
  <div
    data-testid="box"
    width="100"
  >
    <div
      data-testid="box"
      flexdirection="column"
      width="70"
    >
      <span
        data-testid="text"
      >
        Solution file: solution.ts 0 bytes
      </span>
      <span
        data-testid="text"
      >
           Input file: input.txt 0 bytes
      </span>
    </div>
  </div>
</div>
`;

exports[`FileSection renders correctly with file info 1`] = `
<div>
  <div
    data-testid="box"
    width="100"
  >
    <div
      data-testid="box"
      flexdirection="column"
      width="70"
    >
      <span
        data-testid="text"
      >
        Solution file: solution.ts 123 bytes
      </span>
      <span
        data-testid="text"
      >
           Input file: input.txt 456 bytes
      </span>
    </div>
  </div>
</div>
`;
